lookout/style/format/rules.py

```diff
@@ -18,22 +18,45 @@
 from sklearn.tree import _tree as Tree, DecisionTreeClassifier
 
 from lookout.core.ports import Type
+from lookout.style.format.feature_extractor import FeatureGroup, IndexToFeature
+from lookout.style.format.features import Feature, get_features
+
+
+class RuleAttribute(NamedTuple(
+        "RuleAttribute", (("feature", int), ("cmp", bool), ("threshold", float)))):
+    """
+    Hold the feature, comparison operator and threshold of a split.
+
+    `feature` is the feature taken for comparison
+    `cmp` is the comparison type: True is "x > v", False is "x <= v"
+    `threshold` is "v", the threshold value
+    """
+
+    def describe(self, index_to_feature: IndexToFeature, features: Mapping[str, Feature]) -> str:
+        """Describe a rule attribute."""
+        group, repetition, feature_name, index = index_to_feature[self.feature]
+        node_description = ("%s %d" % (group.name, repetition)
+                            if group is not FeatureGroup.node
+                            else "%s" % group.name)
+        return "%s: %s" % (node_description,
+                           features[feature_name].describe_rule_part(self.cmp, self.threshold,
+                                                                     index))
 
-RuleAttribute = NamedTuple(
-    "RuleAttribute", (("feature", int), ("cmp", bool), ("threshold", float)))
-"""
-`feature` is the feature taken for comparison
-`cmp` is the comparison type: True is "x > v", False is "x <= v"
-`threshold` is "v", the threshold value
-"""
 
 RuleStats = NamedTuple("RuleStats", (("cls", int), ("conf", float), ("support", int)))
 """
 `cls` is the predicted class
 `conf` is the rule confidence \\in [0, 1], "1" means super confident
 """
 
-Rule = NamedTuple("RuleType", (("attrs", Tuple[RuleAttribute, ...]), ("stats", RuleStats)))
+
+class Rule (NamedTuple("RuleType", (("attrs", Tuple[RuleAttribute, ...]), ("stats", RuleStats)))):
+    """Hold the attributes and statistics of a rule."""
+
+    def describe(self, index_to_feature: IndexToFeature, features: Mapping[str, Feature]) -> str:
+        """Describe a rule."""
```

Comment:
Format the rule as text. We take features metadata to convert the integer indices to human-readable names.
Delegates most of the work to RuleAttribute.describe().